cicd/docker/algod/setup.py

```diff
@@ -15,32 +15,41 @@
 
 # Shared parameters
 base_parser = argparse.ArgumentParser(add_help=False)
-base_parser.add_argument('--bin-dir', required=True, help='Location to install algod binaries.')
-base_parser.add_argument('--channel', required=True, help='Channel to install, nightly is a good option.')
-base_parser.add_argument('--network-dir', required=True, help='Path to create network.')
 
 subparsers = parser.add_subparsers()
 
-install = subparsers.add_parser('install', parents=[base_parser], help='Install the network.')
-install.add_argument('--network-template', required=True, help='Path to private network template file.')
-install.add_argument('--network-token', required=True, help='Valid token to use for algod/kmd.')
-install.add_argument('--algod-port', required=True, help='Port to use for algod.')
-install.add_argument('--kmd-port', required=True, help='Port to use for kmd.')
+install = subparsers.add_parser('install', parents=[base_parser], help='Install binaries.')
+install.add_argument('--bin-dir', required=True, help='Location to install algod binaries.')
+install.add_argument('--channel', required=True, help='Channel to install, nightly is a good option.')
+install.add_argument('--algod-version', required=False, default='2.0.4', help='Channel to install, nightly is a good option.')
+
+create = subparsers.add_parser('create-network', parents=[base_parser], help='Create test network.')
+create.add_argument('--bin-dir', required=True, help='Location to install algod binaries.')
+create.add_argument('--network-dir', required=True, help='Path to create network in.')
+create.add_argument('--network-template', required=True, help='Path to private network template file.')
+
+configure = subparsers.add_parser('configure-network', parents=[base_parser], help='Configure test network.')
+configure.add_argument('--network-dir', required=True, help='Path to configure network.')
+configure.add_argument('--network-token', required=True, help='Valid token to use for algod/kmd.')
+configure.add_argument('--algod-port', required=True, help='Port to use for algod.')
+configure.add_argument('--kmd-port', required=True, help='Port to use for kmd.')
 
 start = subparsers.add_parser('start', parents=[base_parser], help='Start the network.')
+start.add_argument('--bin-dir', required=True, help='Location to install algod binaries.')
+start.add_argument('--network-dir', required=True, help='Path to create network.')
 
 pp = pprint.PrettyPrinter(indent=4)
 
-def setup_algod(bin_dir, channel):
+def setup_algod(bin_dir, channel, algod_version):
     """
     Download and install algod.
     """
     home = expanduser('~')
     os.makedirs("%s/inst" % home, exist_ok=True)
     print('downloading updater...')
-    url='https://algorand-releases.s3.amazonaws.com/channel/stable/install_stable_linux-amd64_2.0.4.tar.gz'
+    url=f"https://algorand-releases.s3.amazonaws.com/channel/stable/install_stable_linux-amd64_{algod_version}.tar.gz"
```

Comment:
If we wanted to simplify this, we could include the new and improved `update.sh` script in this repo. The new script automatically finds and downloads the latest updater.